src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -9,8 +9,15 @@ namespace Microsoft.CodeAnalysis
     /// <summary>
     /// A class that represents all the arguments necessary to create a new document instance.
     /// </summary>
+    [System.Diagnostics.DebuggerDisplay("{GetDebuggerDisplay()}")]
     public sealed class DocumentInfo
     {
+
+        private string GetDebuggerDisplay()
+        {
+            return nameof(FilePath) + " = " + FilePath;
```

Comment:
DocumentInfo is not required to have a file path.  This property can be null.  It even defaults to null in the factory API.   It is expected to have a name though.  However, looking at the constructor I see that we fail to throw when the name is null.